rolepermissions/roles.py

```diff
@@ -49,24 +49,30 @@ def get_name(cls):
     @classmethod
     def assign_role_to_user(cls, user):
         """
-        Deletes all of user's previous roles, and removes all permissions
-        mentioned in their available_permissions property.
+        Assign this role to a user.
 
         :returns: :py:class:`django.contrib.auth.models.Group` The group for the
             new role.
         """
-        from rolepermissions.shortcuts import remove_role
-
-        remove_role(user)
-
-        group, created = Group.objects.get_or_create(name=cls.get_name())
+        group, _created = Group.objects.get_or_create(name=cls.get_name())
         user.groups.add(group)
         permissions_to_add = cls.get_default_true_permissions()
         user.user_permissions.add(*permissions_to_add)
 
         return group
 
     @classmethod
+    def remove_role_from_user(cls, user):
```

Comment:
Okay so I'm working through this. Keeping explicitly revoked permissions after the role removal is possible, but keeping explicitly granted permissions that are listed as True on the removed role is not (unless I'm missing something).

Two options:

1. Accept the inconsistency and add that case to the documentation
2. Adjust both the ``assign_role`` and ``remove_role`` functions to reset permissions to default when roles are changed. Also remove ``grant_permission`` and ``revoke_permission`` functions and make a point that the default permission sets on Roles should be treated as immutable.

I'm fine with 1., but I think it going with 2. would bring clarity and ensure permissions are truly defined by roles.